## Analysis of Used Car Listings on Ebay

In this analysis, we shall be working on used cars listings from ebay Kleinanzeigen a section of the German eBay website.
This data set was pulled from Kaggle.com. but some modifications done to it by sampling 50000 rows for faster analysis.

The aim of this project is to clean the data and analyze the listings.



In [1]:
#Import libraries needed
import numpy as np
import pandas as pd

In [2]:
#read in file
autos = pd.read_csv("autos.csv",encoding = "Latin-1")

In [3]:
#First four observations of data set
autos.head(4)

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28

In [4]:
#Information about autos data set
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

## View Column Names and Clean Columns

In [5]:
#View column names
all_columns = autos.columns
print(all_columns)

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')


To clean columns, we shall convert the columns from camelcase to snakecase and change a few wordings to make the columns's describtion more accurate. 

In [6]:
all_columns_new = all_columns 

In [7]:
import re

In [8]:
def fixing(var):
        if var == "yearOfRegistration":
            var = "registration_year"
        elif var == "monthOfRegistration":
            var = "registration_month"
        elif var == "notRepairedDamage":
            var = "unrepaired_damage"
        elif var == "dateCreated":
            var = "ad_created"
        else:
            s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', var)
            var = re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
        return var    
        

In [9]:
all_new_columns_2 = []
for i in all_columns_new:
    value = fixing(i)
    all_new_columns_2.append(value)
    
print(all_new_columns_2)    

['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest', 'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model', 'odometer', 'registration_month', 'fuel_type', 'brand', 'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code', 'last_seen']


In [10]:
autos.columns = all_new_columns_2

In [11]:
autos.head(3)

date_crawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2  2016-03-26 18:57:24                  Volkswagen_Golf_1.6_United  privat   

  offer_type   price   abtest vehicle_type  registration_year    gearbox  \
0    Angebot  $5,000  control          bus               2004    manuell   
1    Angebot  $8,500  control    limousine               1997  automatik   
2    Angebot  $8,990     test    limousine               2009    manuell   

   power_ps   model   odometer  registration_month fuel_type       brand  \
0       158  andere  150,000km                   3       lpg     peugeot   
1       286     7er  150,000km                   6    benzin         bmw   
2       102    golf   70,000km                   7    benzin  volkswagen   

  unrepaired_damage           ad_created  nr_of_pictures  postal_code  \
0              nein  2016-03-26 00:00:00               0        79588   
1              nein  2016-04-04 00:00:00               0        71034   
2              nein  2016-03-26 00:00:00               0        35394   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37

## Exploratory Data Analysis 
Here we inspect columns to check anyones that need cleaning. 

In [12]:
autos.describe(include = "all")

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-19 17:36:18  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

The following columns need a closer attention
- We need to convert the date_crawled column to a date
- The name column contains missing value 
- Seller column will be dropped because it contains just two values
- Offer_type column contains just 2 values and will be dropped
- Price column needs to be converted to numeric
- Abtest will be dropped
- Vehicle_type has missing data
- Registration year has inconsistent data, so will be dropped
- Odometer should be transformed to numeric value
- Fuel type contains 7 unique values we need to inspect that
- Unrepaired damage will be dropped
- nr_of_pics should be converted to numeric
- postal code we need to closer look at the distribution
- last seen will be dropped


## Explore Odometer and Price Column

In [13]:
#Inspect price column 
autos["price"].value_counts()


$0         1421
$500        781
$1,500      734
$2,500      643
$1,200      639
$1,000      639
$600        531
$3,500      498
$800        498
$2,000      460
$999        434
$750        433
$900        420
$650        419
$850        410
$700        395
$4,500      394
$300        384
$2,200      382
$950        379
$1,100      376
$1,300      371
$3,000      365
$550        356
$1,800      355
$5,500      340
$1,250      335
$350        335
$1,600      327
$1,999      322
           ... 
$13,049       1
$12,889       1
$4,485        1
$11,448       1
$6,998        1
$33,777       1
$2,410        1
$5,634        1
$7,930        1
$34,980       1
$2,245        1
$3,725        1
$1,169        1
$10,488       1
$10,090       1
$135          1
$21,690       1
$23,450       1
$295          1
$1,039        1
$1,645        1
$9,325        1
$1,765        1
$4,567        1
$14,525       1
$70,000       1
$17,799       1
$789          1
$24,242       1
$14,337       1
Name: price, Length: 235

Again, the prices in this column seem rounded, however given there are 2357 unique values in the column, that may just be people's tendency to round prices on the site.

There are 1,421 cars listed with $0 price - given that this is only 2% of the of the cars, we might consider removing these rows.

In [14]:
#Inspect odometer column closer
autos["odometer"].value_counts()

150,000km    32424
125,000km     5170
100,000km     2169
90,000km      1757
80,000km      1436
70,000km      1230
60,000km      1164
50,000km      1027
5,000km        967
40,000km       819
30,000km       789
20,000km       784
10,000km       264
Name: odometer, dtype: int64

We can see that the values in this field are rounded, which might indicate that sellers had to choose from pre-set options for this field. Additionally, there are more high mileage than low mileage vehicles.

In [15]:
#Remove non-digit special chracters
autos["price"] = autos["price"].str.replace("$"," ")

In [16]:
#Remove non-digit and special chararcter from odometer
autos["odometer"] = autos["odometer"].str.replace("km"," ")

In [17]:
#Rename odometer column
autos.rename({"odometer":"odometer_km"},axis = 1,inplace = True)

In [18]:
#View of new column names
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer_km', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

In [19]:
#Cleaned the price and odometer columns
autos["price"] = autos['price'].str.replace('$','').str.replace(',','').astype(int)
autos["odometer_km"] = autos['odometer_km'].str.replace(',','').str.replace('km','').astype(int)

In [20]:
#Lets view the data types in both columns to affirm conversion
print(autos["price"].dtype)
print(autos["odometer_km"].dtype)

int64
int64


In [21]:
#Lets check on how many unique values are in columns odometer_km and price
print("Unique prices are = ", autos["price"].unique().shape)
print("Unique odometers are = ", autos["odometer_km"].unique().shape)

Unique prices are =  (2357,)
Unique odometers are =  (13,)


In [22]:
#Distribution of price columns
print(autos["price"].describe())

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64


In [23]:
#Distribution of odometer columns
print(autos["odometer_km"].describe())

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64


In [24]:
#Let us view counts of values for price column 
print(autos["price"].value_counts().head(10).sort_index(ascending = False))

3500     498
2500     643
2000     460
1500     734
1200     639
1000     639
800      498
600      531
500      781
0       1421
Name: price, dtype: int64


In [25]:
#Let us view counts of values for odometer column 
print(autos["odometer_km"].value_counts().head(20).sort_index(ascending = False))

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
40000       819
30000       789
20000       784
10000       264
5000        967
Name: odometer_km, dtype: int64


In [26]:
#Lets calculate interquantile range
IQR_price = 720000 - 1100
IQR_odometer = 150000 - 125000

In [27]:
#Calculate upper and lower bound for price and odometer
lower_bound_price = 1100 - (1.5 * IQR_price)
upper_bound_price = 720000 + (1.5 * IQR_price)

lower_bound_odometer = 125000 - (1.5 * IQR_odometer)
upper_bound_odometer = 150000 + (1.5 * IQR_odometer)

In [28]:
print(lower_bound_price,upper_bound_price)
print(lower_bound_odometer, upper_bound_odometer)

-1077250.0 1798350.0
87500.0 187500.0


Values to be considered for upper and lower bounds 
For prices, because the lower bound falls below the lowest value, we shall consider the
lowest value and upper bound which is 0 and 1,798,350


In [29]:
autos = autos.loc[autos["price"].between(0,1798350),:]

What values do we consider for upper and lower bound?
The lower bound will be 87500 and the upper bound will be 150000, this is because the upper limit lies above the max value.

In [30]:
autos = autos.loc[autos["odometer_km"].between(87500,150000),:]

In [31]:
print(autos.describe())

              price  registration_year      power_ps    odometer_km  \
count  4.151400e+04       41514.000000  41514.000000   41514.000000   
mean   4.428849e+03        2002.820759    114.485667  141736.040854   
std    1.192271e+04          35.025253    192.731571   17101.772907   
min    0.000000e+00        1910.000000      0.000000   90000.000000   
25%    9.990000e+02        1999.000000     70.000000  150000.000000   
50%    2.450000e+03        2003.000000    105.000000  150000.000000   
75%    5.500000e+03        2007.000000    147.000000  150000.000000   
max    1.234566e+06        9000.000000  17700.000000  150000.000000   

       registration_month  nr_of_pictures   postal_code  
count        41514.000000         41514.0  41514.000000  
mean             5.718384             0.0  50591.217276  
std              3.731841             0.0  25620.497673  
min              0.000000             0.0   1067.000000  
25%              3.000000             0.0  30179.000000  
50%         

## Exploring Date columns
There are a number of columns with date information:

- date_crawled
- registration_month
- registration_year
- ad_created
- last_seen

These are a combination of dates that were crawled, and dates with meta-information from the crawler. The non-registration dates are stored as strings.

We'll explore each of these columns to learn more about the listings.

In [32]:
#Lets do some exploration in the date_crawled column
print(autos["date_crawled"].str[:10].value_counts(normalize = True,dropna = False)
      .sort_index(ascending = False))

2016-04-07    0.001204
2016-04-06    0.003445
2016-04-05    0.013514
2016-04-04    0.036735
2016-04-03    0.038108
2016-04-02    0.035000
2016-04-01    0.033459
2016-03-31    0.031989
2016-03-30    0.033820
2016-03-29    0.033820
2016-03-28    0.034711
2016-03-27    0.031218
2016-03-26    0.032326
2016-03-25    0.032399
2016-03-24    0.028930
2016-03-23    0.032543
2016-03-22    0.032808
2016-03-21    0.037554
2016-03-20    0.037626
2016-03-19    0.033868
2016-03-18    0.013249
2016-03-17    0.031965
2016-03-16    0.029532
2016-03-15    0.033964
2016-03-14    0.036855
2016-03-13    0.014983
2016-03-12    0.037048
2016-03-11    0.032206
2016-03-10    0.032182
2016-03-09    0.033892
2016-03-08    0.034109
2016-03-07    0.036205
2016-03-06    0.014092
2016-03-05    0.024642
Name: date_crawled, dtype: float64


Based on the distribution above, most of the data was crawled in the month of March


In [33]:
#Lets do some exploration in the ad_created column
print(autos["ad_created"].str[:10].value_counts(normalize = True,dropna = False)
      .sort_index(ascending = False))

2016-04-07    0.001108
2016-04-06    0.003469
2016-04-05    0.012116
2016-04-04    0.037289
2016-04-03    0.038373
2016-04-02    0.034591
2016-04-01    0.033531
2016-03-31    0.031989
2016-03-30    0.033603
2016-03-29    0.033964
2016-03-28    0.034663
2016-03-27    0.031146
2016-03-26    0.032278
2016-03-25    0.032567
2016-03-24    0.028834
2016-03-23    0.032375
2016-03-22    0.032856
2016-03-21    0.037698
2016-03-20    0.037602
2016-03-19    0.032880
2016-03-18    0.013947
2016-03-17    0.031700
2016-03-16    0.029990
2016-03-15    0.033748
2016-03-14    0.035241
2016-03-13    0.016500
2016-03-12    0.036855
2016-03-11    0.032543
2016-03-10    0.032013
2016-03-09    0.033699
                ...   
2016-02-27    0.000120
2016-02-26    0.000024
2016-02-25    0.000048
2016-02-24    0.000048
2016-02-23    0.000024
2016-02-22    0.000024
2016-02-21    0.000072
2016-02-20    0.000048
2016-02-19    0.000024
2016-02-18    0.000048
2016-02-16    0.000024
2016-02-14    0.000024
2016-02-12 

Based on the above, most of the ad were created in the month of March 2016.A very small proportion was created in 2015 which could be considered outlier.

In [34]:
#Lets do some exploration in the last_seen column
print(autos["last_seen"].str[:10].value_counts(normalize = True,dropna = False)
      .sort_index(ascending = False))

2016-04-07    0.125379
2016-04-06    0.212579
2016-04-05    0.119719
2016-04-04    0.024907
2016-04-03    0.025509
2016-04-02    0.025534
2016-04-01    0.023992
2016-03-31    0.024401
2016-03-30    0.024883
2016-03-29    0.022860
2016-03-28    0.021824
2016-03-27    0.016621
2016-03-26    0.017392
2016-03-25    0.020090
2016-03-24    0.020523
2016-03-23    0.019150
2016-03-22    0.022113
2016-03-21    0.021559
2016-03-20    0.021824
2016-03-19    0.016525
2016-03-18    0.007347
2016-03-17    0.029364
2016-03-16    0.017103
2016-03-15    0.016308
2016-03-14    0.013224
2016-03-13    0.009370
2016-03-12    0.025076
2016-03-11    0.013465
2016-03-10    0.011081
2016-03-09    0.010334
2016-03-08    0.008335
2016-03-07    0.005829
2016-03-06    0.004697
2016-03-05    0.001084
Name: last_seen, dtype: float64


The distribution of the data shows that the crawler was most present online in the month of March 2016.Although April 2016 shows only 4 months, the crawler was present 45% of the time.

In [35]:
autos["registration_year"].describe()

count    41514.000000
mean      2002.820759
std         35.025253
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2007.000000
max       9000.000000
Name: registration_year, dtype: float64

The max value for this column is 9000 and we are not yet in the year 9000. 

## Dealing with Incorrect Year in registration_year Column

The min and max value will be 1910 because the first cars were invented in 1885 and max will be 2016 because we cannot make cars pass the year of registration.
For this reason, we shall select rows of car between the year 1910 and 2016 from our data set.

In [36]:
autos = autos.loc[autos["registration_year"].between(1910,2016),:]

In [37]:
autos["registration_year"].value_counts(normalize = True).sort_index()

1910    0.000050
1934    0.000025
1937    0.000025
1943    0.000025
1950    0.000025
1953    0.000025
1954    0.000050
1955    0.000025
1956    0.000050
1957    0.000025
1958    0.000050
1959    0.000075
1960    0.000402
1961    0.000075
1962    0.000025
1963    0.000126
1964    0.000075
1965    0.000226
1966    0.000352
1967    0.000327
1968    0.000327
1969    0.000226
1970    0.000427
1971    0.000176
1972    0.000528
1973    0.000477
1974    0.000327
1975    0.000226
1976    0.000452
1977    0.000377
          ...   
1987    0.001659
1988    0.002991
1989    0.003920
1990    0.008218
1991    0.008193
1992    0.008896
1993    0.010455
1994    0.015355
1995    0.030057
1996    0.034279
1997    0.048931
1998    0.058305
1999    0.071850
2000    0.078460
2001    0.064487
2002    0.060265
2003    0.064738
2004    0.064462
2005    0.068985
2006    0.061094
2007    0.050564
2008    0.045840
2009    0.036943
2010    0.024578
2011    0.019427
2012    0.009751
2013    0.002513
2014    0.0003

From the above distribution,most the cars were registered from 1993 to 2011

## Exploring Price by Brand

In [38]:
autos["brand"].value_counts(normalize = True)

volkswagen        0.217486
bmw               0.117413
opel              0.113166
mercedes_benz     0.099520
audi              0.088613
ford              0.069036
renault           0.049383
peugeot           0.030107
fiat              0.023925
seat              0.017491
mazda             0.015179
nissan            0.014199
skoda             0.013722
citroen           0.013219
toyota            0.011284
smart             0.010555
volvo             0.010304
mitsubishi        0.008318
hyundai           0.007816
honda             0.007615
sonstige_autos    0.007162
alfa_romeo        0.007087
kia               0.006157
mini              0.005353
suzuki            0.005051
porsche           0.004549
chevrolet         0.003971
chrysler          0.003971
daihatsu          0.002438
jeep              0.002237
subaru            0.002111
saab              0.001860
land_rover        0.001835
jaguar            0.001533
rover             0.001483
dacia             0.001483
daewoo            0.001458
l

I will explore variation across various car brands and will be using the top 20 cars based on percentage value.

In [39]:
top_autos = autos["brand"].value_counts(normalize = True).head(20).index
print(top_autos)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat', 'mazda', 'nissan', 'skoda', 'citroen',
       'toyota', 'smart', 'volvo', 'mitsubishi', 'hyundai', 'honda'],
      dtype='object')


In [40]:
brand_dict = {}

for i in top_autos:
    rows = autos[autos["brand"] == i]
    mean_price = rows["price"].mean()
    brand_dict[i] = int(mean_price) 

sort_list = sorted(brand_dict.items(),key=lambda x: x[1],reverse = True)  

for i in sort_list:
    print(i[0],i[1])

bmw 6931
audi 6753
mercedes_benz 6691
skoda 4808
volvo 4233
volkswagen 4217
toyota 3940
hyundai 3620
nissan 3133
honda 3008
ford 2950
citroen 2932
seat 2918
mazda 2797
mitsubishi 2503
smart 2473
peugeot 2358
opel 2217
fiat 1891
renault 1779


Based on the average prices computed, we can see that BMW, Audi and Mercedes top the list in the range of 6k , the closest group that follows is Volvo,Volkswagen and Skoda in the range of 4k.

In [41]:
top_six = sort_list[:6]
print(top_six)

[('bmw', 6931), ('audi', 6753), ('mercedes_benz', 6691), ('skoda', 4808), ('volvo', 4233), ('volkswagen', 4217)]


## Exploring Mean Price and Mean Mileage by Brand

In [42]:
#Lets focus on the top brands and calculate their average price and mileage
top_brands = ["bmw","audi","mercedes_benz","skoda","volvo","volkswagen"]

price_dict = {}
mileage_dict = {}

for i in top_brands:
    rows = autos[autos["brand"] == i]
    mean_price = rows["price"].mean()
    price_dict[i] = int(mean_price)
    
print(price_dict)
print("        ")

for i in top_brands:
    rows = autos[autos["brand"] == i]
    mean_mileage = rows["odometer_km"].mean()
    mileage_dict[i] = int(mean_mileage)
    
print(mileage_dict)    

{'bmw': 6931, 'skoda': 4808, 'mercedes_benz': 6691, 'volkswagen': 4217, 'volvo': 4233, 'audi': 6753}
        
{'bmw': 143590, 'skoda': 136730, 'mercedes_benz': 143613, 'volkswagen': 143286, 'volvo': 145682, 'audi': 144144}


In [43]:
#Create series objects
price_series = pd.Series(price_dict)

mileage_series = pd.Series(mileage_dict)

#Create dataframe
df = pd.DataFrame({"mean_price" : price_series,"mean_mileage" : mileage_series})
print(df)

               mean_mileage  mean_price
audi                 144144        6753
bmw                  143590        6931
mercedes_benz        143613        6691
skoda                136730        4808
volkswagen           143286        4217
volvo                145682        4233


Based on the information above,audi cars sell at a relatively high price with a high mean_mileage value.Mercedes cars as the most expensive with mileage over 140000km.We can also observe that volkswagen cars have a realtively lower mileage and lower mean price.